src/axom/quest/interface/signed_distance.cpp

```diff
@@ -383,6 +383,64 @@ double signed_distance_evaluate(double x, double y, double z)
   return (phi);
 }
 
+//------------------------------------------------------------------------------
+double signed_distance_evaluate(double x,
+                                double y,
+                                double z,
+                                double& cp_x,
+                                double& cp_y,
+                                double& cp_z,
+                                double& n_x,
+                                double& n_y,
+                                double& n_z)
```

Comment:
Note: The `SignedDistance` class supports returning the closest points and surface normals for a batch of points, but in the C-style API, I only created and overload for the query over a single point, since:
(a) this is the one that @dtaller is planning to use 
(b) creating one for the latter would require either copying data from an arrays of point/normal components to an array of points/vectors OR extending the `ZipIterator` interface to support writing data. The latter felt out of scope for this PR